### DISCUSSION TITLE
6 OpenDTU-Einheiten mit einem MQTT Broker verbinden möglich?

### DISCUSSION THREAD

**Discussion by HaraldNoack** [tbnobody/OpenDTU#384] [General]
Hallo,

DANKE für dieses Projekt!

Ist es möglich 6 OpenDTU-Einheiten mit einem MQTT Broker zu verbinden?

Ich möchte mehrere Inverter auf 6 OpenDTU-Einheiten verteilen um 6 in der Leistung unabhängig steuerbare Gruppen zu bilden.
(Phasenaufteilung usw..) 

Ist dieses wegen des "General topics"-Namens "dtu/" möglich?
Ist es möglich diesen zu ändern zu z.B.: "dtu1/" , "dtu2/" , "dtu3/"

Ist es möglich die Leistung aller an eine OpenDTU-Einheit angeschlossenen Inverter über einen Befehl zu ändern? 
( 5 Inverter an einer OpenDTU-Einheit und es soll eine Null-Einspeisung für diese Phase erzeugt werden. 
Dies jetzt 3x für 3 Phasen. Das selbe jetzt noch 3 mal für den Batteriespeicher) 

VIELEN DANK FÜR EURE HILFE !!!

**Comment by joba-1** [comment]
ja, das ist möglich. Angenommen eine deiner DTU hat den Hostnamen opendtu1, dann kannst du der unter der url http://opendtu1/settings/mqtt ein anderes Base Topic geben. Denkbar wäre z.B. OpenDTU/1 und das Subtopic dtu könnte man dann mit  OpenDTU/1/dtu abfragen.

Wenn du dann noch eine Möglichkeit brauchst, die Daten in einer Influx-Datenbank zu sammeln, kannst du in 5 Minuten hier nachschauen: https://github.com/joba-1/openDtu2Db - bin gerade am hochladen :)

**Reply by dirkheiden** [reply]
Warum nutzt Du nicht Telegraf für die Verbindung von MQTT und Influx?

**Reply by joba-1** [reply]
Naja, das hatte vor allem einen Grund: Ich kannte es nicht.

Hab es mir jetzt (kurz) angeschaut. Es gibt es noch mehr Gründe, warum ich gern bei Python bleibe:

### OpenDTU schickt Einzelwerte, In Influx will ich sie in Measurements gesammelt haben

Ja, das ist in Telegraf offenbar auch möglich, aber die Config wird dann schon etwas komplexer.
Wann eine Gruppe von Daten vollständig ist, ist mir noch nicht vollständig klar bzw. hängt vom Device und der DTU Firmware ab und kann sich auch ändern: Ich muss das also vermutlich noch anpassen. Ob das mit Telegraf Konfiguration dann noch geht ist offen - mit Python geht es auf jeden Fall irgendwie.

### Manche Werte will ich nicht 1:1 übernehmen sondern verarbeiten

z.B. soll die sich ständig ändernde Uptime als eine feste Starttime gespeichert werden. Geht das mit Telegraf? So weit bin ich jetzt nicht eingestiegen. Aber selbst wenn, so komplex ist der Python Code insgesamt nicht, das kommt dann schon vom Aufwand in ähnliche Regionen.

### Telegraf ist eine Speziallösung

Wenn ich die obigen Probleme angehe und Telegraf lerne, kann ich in Zukunft MQTT -> Influx Probleme damit lösen. Wenn ich in Python weiter komme, kann ich viel mehr Probleme lösen - es ist das universellere Werkzeug.
Außerdem (bzw. aus diesem Grund) ist Python ohnehin auf fast jedem meiner Systeme im Einsatz, Telegraf hat vermutlich eher noch niemand, muss man also erst mal installieren.

### Die Influx API hat sich schnell recht radikal geändert

Schon der Connect mit früher geschriebenem Code mit Influx Client Python Libs hat nicht funktioniert - schon die "richtige" zu finden hat gedauert. Wenn ich das auf Telegraf übertrage (kommt ja von den selben Leuten), dann riskiere ich, dass ich mit dem was ich lerne in 5 Jahren gar nichts mehr anfange.
Mein Python und SQL Wissen von vor 10 Jahren ist dagegen immer noch sehr brauchbar...

Ich überlasse es daher gern professionellen MQTT -> Influx Gateway Programmierern sich in Telegraf einzuarbeiten - da lohnt der Aufwand vermutlich :)
